src/lib/components/SegmentButton.svelte

```diff
@@ -17,32 +18,45 @@
 
   let selected = false;
   $: selected = $store.id === segmentId;
+
+  // Update the store with initially selected element as soon as it is available
```

Comment:
The code below updates continuously. Should it only update once on init? That's what the comment seems to say.

Reply:
Sorry, I didn't get this. For me this block updates the store only once.

Reply:
This code in a `$:` block is re-evaluated every time any of the values changes.
So every time `selected` or `$store` changes, if the condition holds, it will set the store value. Is this intended?
Or are you saying that these values never change after they get their initial value?

Reply:
Ah, got your point. The issue this solves is that initially there is no `element` in the store https://github.com/dfinity/gix-components/blob/6952fff6a7099cdd367a850a6bee0881136cc8e5/src/lib/components/Segment.svelte#L12.
But it's already there after clicking on a button, so every time the condition is re-evaluated `$store.element !== element` is false.

Reply:
But we need an element in the store to be able to set the right indicator position. This was a bug of the component before the redesign. Now we need this functionality.